liquidator/index.js

```diff
@@ -57,6 +60,34 @@ async function run(address, shouldPoll, pollingDelay, priceFeedConfig, monitorPo
     const gasEstimator = new GasEstimator(Logger);
     const liquidator = new Liquidator(Logger, empClient, gasEstimator, priceFeed, accounts[0]);
 
+    // The EMP requires approval to transfer the liquidator's collateral and synthetic tokens in order to liquidate
+    // a position. We'll set this once to the max value and top up whenever the bot's allowance drops below
+    // MAX_INT / 2.
+    const collateralToken = await ExpandedERC20.at(await emp.collateralCurrency());
+    const syntheticToken = await ExpandedERC20.at(await emp.tokenCurrency());
+    const currentCollateralAllowance = await collateralToken.allowance(accounts[0], empClient.empAddress);
+    const currentSyntheticAllowance = await syntheticToken.allowance(accounts[0], empClient.empAddress);
+    if (toBN(currentCollateralAllowance).lt(toBN(MAX_UINT_VAL).div(toBN("2")))) {
+      const collateralApprovalTx = await collateralToken.approve(empClient.empAddress, MAX_UINT_VAL, {
+        from: accounts[0]
+      });
+      Logger.debug({
```

Comment:
I think this should be an `info` level log. The bot is making actual on-chain transactions here. It would be nice to know about it in the slack channel.

Reply:
+1

Reply:
done